dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -585,11 +607,114 @@ public void Test_Required_Params() throws IOException, DotDataException, DotSecu
+                "lola",
+                "A bunch of string params to demo the mechanism.", false, paramMap)) {
+
+            // Create Service integration Descriptor
+            final Response serviceIntegrationResponse = serviceIntegrationResource
+                    .createServiceIntegration(request, response,
+                            createFormDataMultiPart(fileName, inputStream));
+            Assert.assertNotNull(serviceIntegrationResponse);
+            Assert.assertEquals(HttpStatus.SC_OK, serviceIntegrationResponse.getStatus());
+            final Response availableServicesResponse = serviceIntegrationResource
+                    .listAvailableServices(request, response);
+            Assert.assertEquals(HttpStatus.SC_OK, availableServicesResponse.getStatus());
+            final ResponseEntityView responseEntityView1 = (ResponseEntityView) availableServicesResponse
+                    .getEntity();
+            final List<ServiceIntegrationView> integrationViewList = (List<ServiceIntegrationView>) responseEntityView1
+                    .getEntity();
+            Assert.assertFalse(integrationViewList.isEmpty());
+            Assert.assertTrue(
+                    integrationViewList.stream().anyMatch(
+                            serviceIntegrationView -> "lola"
+                                    .equals(serviceIntegrationView.getName())));
+
+            // Add secrets to it.
+            for(final Host host:hosts) {
+                createSecret(request, response, serviceKey, host, paramMap);
+            }
+
+            //fetch and test pagination
+            final int pageSize = 4;
+            final int numberOfPages = hosts.size() / pageSize;
+
+            for(int currentPage = 1; currentPage <= numberOfPages; currentPage++) {
+
+                final Response paginationResponse1 = serviceIntegrationResource
+                        .getServiceIntegrationByKey(request, response, serviceKey,
+                                new PaginationContext(null, currentPage, pageSize, "", ""));
+                Assert.assertEquals(HttpStatus.SC_OK, paginationResponse1.getStatus());
+                final ResponseEntityView paginationEntity1 = (ResponseEntityView) paginationResponse1
+                        .getEntity();
+                final ServiceIntegrationView paginationView1 = (ServiceIntegrationView) paginationEntity1
+                        .getEntity();
+                Assert.assertEquals(hosts.size(), paginationView1.getConfigurationsCount());
+                final List<SiteView> hostsBatch1 = paginationView1.getSites();
+                Assert.assertNotNull(hostsBatch1);
+
+                if(currentPage < numberOfPages) {
+                    Assert.assertEquals(pageSize, hostsBatch1.size());
+                } else {
+                    Assert.assertTrue(pageSize <= hostsBatch1.size());
+                }
+
+                List<SiteView> itemsPage;
+                if(currentPage == 1){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960082)